chroot_setup

```diff
@@ -0,0 +1,440 @@
+        os.chmod(os.path.join(root, file), int(mode, 8))
+
+# :chown
+# Change ownership of a file in the chroot
+def chroot_chown (line: str):
+  chown_command = line.split()
+  assert len(chown_command) == 2, ":chown command expect 2 arguments!"
+  user = chown_command[0]
+  chown_file = args.chroot_directory + chown_command[1]
+  printf(f"os.chown({args.chroot_directory + chown_file}, {user})")
+  if (CHROOT_DEBUG == False):
+    os.chown(args.chroot_directory + chown_file, user)
+
+# :chown_dir
+# Change ownership of a directory and everything inside it in the chroot.
+def chroot_chown_dir (line: str):
+  chown_command = line.split()
+  assert len(chown_command) == 2, ":chown_dir command expect 2 arguments!"
+  user = chown_command[0]
+  chown_directory = args.chroot_directory + chown_command[1]
+  for root, dirs, files in os.walk(args.chroot_directory + chown_directory):
+    for dir in dirs:
+      printf(f'os.chown({os.path.join(root, dir)}, {user})')
+      if (CHROOT_DEBUG == False):
+        os.chown(os.path.join(root, dir), user)
+    for file in files:
+      printf(f'os.chown({os.path.join(root, file)}, {user})')
+      if (CHROOT_DEBUG == False):
+        os.chown(os.path.join(root, file), user)
+
+# :system
+# Executes a line from a .lst file using the /QOpenSys/usr/bin/system utility
+def chroot_system(line: str):
+  line = line.rstrip()
+  print('system -i "' + line + '"')
+  if (CHROOT_DEBUG == False):
+    os.system('system -i "' + line + '"')
+
+# :sh
+# Executes a line from a .lst file using Pythons os.system function, which is
+# run "by calling the Standard C function system()."
+def chroot_sh(line: str):
+  print(line)
+  if (CHROOT_DEBUG == False):
+    os.system(line)
+
+# :file
+# Parses and executes additional .lst files
+def chroot_setup (chroot_lst):
+  lst_file = open(chroot_lst, 'r')
+  action = None
+
+  # Iterate over all lines in the file, executing actions where encountered
+  for line in lst_file:
+    line = line.strip()
+
+    # Skip blank lines and comments
+    if not line or line.startswith('#'):
+      continue
+
+    # Determine action
+    elif line.startswith(':'):
+      action = action_dictionary[line]
+
+    # Execute action
+    else:
+      if action is not None:
+        action(line)
+
+# Mapping the .lst action keys to the functions that run on their values
+action_dictionary = {
+  ':file'       : chroot_setup,
+  ':mkdir'      : chroot_mkdir,      # done
+  ':ln_fix_rel' : chroot_ln_fix_rel, # done
+  ':ln_rel'     : chroot_ln_rel,     # done
+  ':ln'         : chroot_ln,         # done
+  ':mknod'      : chroot_mknod,      # done
+  ':cp_dir'     : chroot_cp_dir,     # done
+  ':cp'         : chroot_cp,         # done
+  ':chmod_dir'  : chroot_chmod_dir,  # done
+  ':chmod'      : chroot_chmod,      # done
+  ':chown_dir'  : chroot_chown_dir,  # done
+  ':chown'      : chroot_chown,      # done
+  ':system'     : chroot_system,     # done
+  ':sh'         : chroot_sh          # done
+}
```

Comment:
If we wanted to enforce a consistent function naming scheme, this mapping could go away:

```python
    elif line.startswith(':'):
        function = f"chroot_{line[1:]}"
        action = globals()[function]
```

In addition, it might be good to catch the KeyError if the user gives an unexpected action.

Reply:
The only inconsistent action -> function name is:

```
':file'       : chroot_setup,
```

We could make a wrapper `chroot_file` function that calls `chroot_setup`

Reply:
Fixed up in 
ca5966b